app/services/acceptable_verified_credential_adoption_service.rb

```diff
@@ -24,18 +24,23 @@ def result
 
   def display_organic_modal_for_logingov_conversion
     result[:organic_modal] =
-      Flipper.enabled?(:organic_dsl_conversion_experiment, user) && user_qualifies_for_conversion?
+      Flipper.enabled?(:organic_conversion_experiment, user) && user_qualifies_for_conversion?
+    result[:service_name] = user.identity.sign_in[:service_name]
     result
   end
 
   def user_qualifies_for_conversion?
-    logged_in_with_dsl? && !verified_credential_at?
+    (logged_in_with_dsl? || logged_in_with_mhv?) && !verified_credential_at?
   end
 
   def logged_in_with_dsl?
     user.identity.sign_in[:service_name] == SAML::User::DSLOGON_CSID
   end
 
+  def logged_in_with_mhv?
+    user.identity.sign_in[:service_name] == SAML::User::MHV_ORIGINAL_CSID
```

Comment:
You know, I went back and forth on that for a while. But since you're suggesting it too, I think I will go with the memoized version. I'll get that change pushed up shortly. Thanks!